tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/annotations/IconAnnotationTest.java

```diff
@@ -51,19 +55,22 @@ public void setup() {
     public void testIconOnFunction() {
         BLangFunction fooFunction = (BLangFunction) ((List) ((BLangPackage) result.getAST()).functions).get(0);
         BLangAnnotationAttachment annot = (BLangAnnotationAttachment) ((List) fooFunction.annAttachments).get(0);
-        Assert.assertEquals(annot.expr.toString(), " {path: /fooIconPath.icon}");
+        Assert.assertEquals(getActualExpressionFromAnnotationAttachmentExpr(annot.expr).toString(), " {path: " +
+                "/fooIconPath.icon}");
     }
 
     @Test
     public void testIconOnObjectAndMemberFunction() {
         TypeDefinition objType = result.getAST().getTypeDefinitions().get(0);
         List<? extends AnnotationAttachmentNode> objAnnot = objType.getAnnotationAttachments();
         Assert.assertEquals(objAnnot.size(), 1);
-        Assert.assertEquals(objAnnot.get(0).getExpression().toString(), " {path: /barIconPath.icon}");
+        Assert.assertEquals(objAnnot.get(0).getExpression().toString(), " {path: /barIconPath.icon}.cloneReadOnly()");
```

Comment:
Updated new test cases. Need a helper function later to get the exact expression value.